bindings.c

```diff
@@ -4448,6 +4450,82 @@ void *load_begin(void *arg)
 	}
 }
 
+static int proc_loadavg_read(char *buf, size_t size, off_t offset,
+		struct fuse_file_info *fi)
+{
+	struct fuse_context *fc = fuse_get_context();
+	struct file_info *d = (struct file_info *)fi->fh;
+	char *cg;
+	size_t total_len = 0;
+	char *cache = d->buf;
+	struct load_node *n;
+	int hash;
+	int cfd;
+	unsigned long a, b, c;
+
+	if (offset) {
+		if (offset > d->size)
+			return -EINVAL;
+		if (!d->cached)
+			return 0;
+		int left = d->size - offset;
+		total_len = left > size ? size : left;
+		memcpy(buf, cache + offset, total_len);
+		return total_len;
+	}
+
+	pid_t initpid = lookup_initpid_in_store(fc->pid);
+	if (initpid <= 0)
+		initpid = fc->pid;
+	cg = get_pid_cgroup(initpid, "cpu");
+	if (!cg)
+		return read_file("/proc/loadavg", buf, size, d);
+
+	prune_init_slice(cg);
+	hash = calc_hash(cg);
+	n = locate_node(cg, hash);
+
+	/* first time */
+	if (n == NULL) {
+		if (!find_mounted_controller("cpu", &cfd)) {
+			pthread_rwlock_unlock(&load_hash[hash]->rdlock);
```

Comment:
Where is the corresponding lock taken? It's at least not obvious so a comment would be very helpful so that people don't get confused.

Reply:
OK, i will add it.
In locate_node() above, i don't use pthread_rwlock_unlock() because delete is not allowed before read has ended.